akka-stream-tests/src/test/scala/akka/stream/scaladsl/LazySinkSpec.scala

```diff
@@ -86,22 +85,21 @@ class LazySinkSpec extends StreamSpec {
 
     "failed gracefully when upstream failed" in assertAllStagesStopped {
       val sourceProbe = TestPublisher.manualProbe[Int]()
-      val futureProbe = Source.fromPublisher(sourceProbe).runWith(Sink.lazyInit[Int, Probe[Int]](_ ⇒ Future.successful(TestSink.probe[Int]), fallback))
+      val futureProbe = Source.fromPublisher(sourceProbe).runWith(Sink.lazyInitAsync(() ⇒ Future.successful(TestSink.probe[Int])))
 
       val sourceSub = sourceProbe.expectSubscription()
       sourceSub.expectRequest(1)
       sourceSub.sendNext(0)
-      val probe = Await.result(futureProbe, 300.millis)
+      val probe = Await.result(futureProbe, 300.millis).get
```

Comment:
I know this was not introduced by you, but perhaps you can follow up in another PR.
No reason for using such tight timeouts here (and several other places), use `remainingOrDefault` instead.
see https://github.com/akka/akka/blob/master/CONTRIBUTING.md#preferred-ways-to-use-timeouts-in-tests

Reply:
Addressing this in follow up as well then